docs/dictionary.rst

```diff
@@ -0,0 +1,1101 @@
+
+GitHub App registration
+-----------------------
+	The recommended way to register a :ref:`dict tool` or :ref:`dict workflow` on Dockstore. This involves creating a :ref:`dict .dockstore.yml` file on the GitHub repository (other source-control methods are not supported) that hosts the tool or workflow, as well as installing the Dockstore GitHub App. This allows a Dockstore entry to remain in sync with the source-control repository automatically, including new branches, tagged commits, and releases created on GitHub after registration of the entry.  
+
+.. note:: This term as we define it here is associated with Dockstore and may have different definitions in other contexts.  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict GitHub App tool:
+
+GitHub App tool
+---------------
+	A :ref:`dict tool` registered using the Dockstore GitHub App.  
+
+.. note:: This term as we define it here is associated with Dockstore and may have different definitions in other contexts.  
+
+see also :ref:`dict GitHub App registration`  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict GitHub App workflow:
+
+GitHub App workflow
+-------------------
+	A :ref:`dict workflow` registered with the Dockstore GitHub App.  
+
+.. note:: This term as we define it here is associated with Dockstore and may have different definitions in other contexts.  
+
+see also :ref:`dict GitHub App registration`  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict immutable:
+
+immutable
+---------
+	Unchanging, unable to be modified. Immutability implies that an object cannot be updated.  
+
+
+.. updated 2022-05-16  
+
+
+
+.. _dict interoperable:
+
+interoperable
+-------------
+	The ability of data or tools from multiple resources to effectively integrate data, or operate processes, across all systems with a moderate degree of effort.  
+
+
+.. updated 2022-05-16  
+
+
+
+.. _dict JSON:
+
+JSON
+----
+pronounced "jason"  
+
+*abbreviation for* JavaScript Object Notation  
+
+	A human-readable file format that originated in JavaScript, but is now used by a variety of applications. Dockstore supports the inclusion of JSON and :ref:`dict YAML` files in entries to provide sample inputs for workflow and tool entries. Some workflow executors, such as :ref:`dict Cromwell`, can use these files to configure their inputs rather than having to manually listing every input when calling the workflow on the command line.  
+
+see also :ref:`dict YAML`  
+
+Further reading: `<https://www.json.org/json-en.html>`_  
+
+.. updated 2022-05-16  
+
+
+
+.. _dict Jupyter:
+
+Jupyter
+-------
+pronounced "Jupiter" like the planet  
+
+	A project focused on developing "notebooks" for programming languages, most famously Python due to it starting as a splinter of iPython in the early 2010s, but including other languages as well such as R. Jupyter notebooks allow for blocks of code to be nestled between markdown text, allowing for easy documentation of the code blocks and reproducibility of analysis.  
```

Comment:
Nitpicking here. The first sentence here seemed a bit bulky to me. Suggested reframing:

A project focused on developing "notebooks" for programming languages, most famously for Python due to it starting as a splinter of iPython in the early 2010s. But Jupyter notebooks are used for other languages as well, such as R.